core/commands/mfs.go

```diff
@@ -0,0 +1,819 @@
+package commands
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	gopath "path"
+	"strings"
+
+	key "github.com/ipfs/go-ipfs/blocks/key"
+	cmds "github.com/ipfs/go-ipfs/commands"
+	dag "github.com/ipfs/go-ipfs/merkledag"
+	mfs "github.com/ipfs/go-ipfs/mfs"
+	path "github.com/ipfs/go-ipfs/path"
+	ft "github.com/ipfs/go-ipfs/unixfs"
+
+	context "github.com/ipfs/go-ipfs/Godeps/_workspace/src/golang.org/x/net/context"
+)
+
+type mfsMountListing struct {
+	Mounts []mfs.RootListing
+}
+
+var MfsCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Manipulate mutable filesystems",
+		ShortDescription: `
+Mfs is an API for manipulating ipfs objects as if they were a unix filesystem.
+They can be seeded with an initial root hash, or by default are an empty directory.
+
+NOTICE:
+This API is currently experimental, likely to change, and may potentially be
+unstable. This notice will be removed when that is no longer the case. Feedback
+on how this API could be improved is very welcome.
+		`,
+	},
+	Options: []cmds.Option{
+		cmds.StringOption("s", "session", "the name of the filesystem to operate on (default=local)"),
```

Comment:
i think `"name of the editing session"` is clearer. naming a filesystem is a bit confusing. Also pls explain some about sessions + naming them in the `ShortDescription` -- even if WIP, doc helps a lot